gcsfs/core.py

```diff
@@ -835,10 +835,12 @@ async def _glob(self, path, prefix="", **kwargs):
             prefix = path[:ind].split("/")[-1]
         return await super()._glob(path, prefix=prefix, **kwargs)
 
-    async def _ls(self, path, detail=False, prefix="", versions=False, **kwargs):
+    async def _ls(self, path, detail=False, refresh=False, prefix="", versions=False, **kwargs):
         """List objects under the given '/{bucket}/{prefix} path."""
         path = self._strip_protocol(path).rstrip("/")
 
+        if refresh:
+            self.invalidate_cache()
```

Comment:
This removes all cached paths, but probably we only wanted to remove this specific one? 
```suggestion
            self.invalidate_cache(path)
```